backend/cmd/kubeconfig.go

```diff
@@ -123,3 +128,127 @@ func GetOwnContext(config *HeadlampConfig) (*Context, error) {
 
 	return &Context{cluster.Name, *cluster, nil}, nil
 }
+
+// getContextFromKubeConfigs returns the contexts from the kubeconfig files.
+func getContextFromKubeConfigs(path string) []Context {
+	var contexts []Context
+
+	if path != "" {
+		delimiter := ":"
+		if runtime.GOOS == "windows" {
+			delimiter = ";"
+		}
+
+		kubeConfigs := strings.Split(path, delimiter)
+		for _, kubeConfig := range kubeConfigs {
+			kubeConfig, err := absPath(kubeConfig)
+			if err != nil {
+				log.Printf("Failed to resolve absolute path of :%s, error: %v\n", kubeConfig, err)
+				continue
+			}
+
+			contextsFound, err := GetContextsFromKubeConfigFile(kubeConfig)
+			if err != nil {
+				log.Println("Failed to get contexts from", kubeConfig, err)
+			}
+
+			contexts = append(contexts, contextsFound...)
+		}
+	}
+
+	return contexts
+}
+
+// refreshHeadlampConfig refreshes the headlamp config.
+// it removes all the contexts that are loaded from kube config and adds the new ones.
+func refreshHeadlampConfig(config *HeadlampConfig) {
+	path := config.kubeConfigPath
+	// load configs
+	contexts := getContextFromKubeConfigs(path)
+	// removing the old contexts from kube config
+	for key, contextProxy := range config.contextProxies {
+		if contextProxy.source == KubeConfig {
+			log.Printf("Removing cluster %q from contextProxies\n", key)
+			delete(config.contextProxies, key)
+		}
+	}
+	// adding the new contexts from kube config
+	for _, context := range contexts {
+		context := context
+		log.Printf("Setting up proxy for context %s\n", context.Name)
+
+		proxy, err := config.createProxyForContext(context)
+		if err != nil {
+			log.Printf("Error setting up proxy for context %s: %s\n", context.Name, err)
+			continue
+		}
+
+		fmt.Printf("\tlocalhost:%d%s%s/{api...} -> %s\n", config.port, config.baseURL, "/clusters/"+context.Name,
+			*context.cluster.getServer())
+
+		config.contextProxies[context.Name] = contextProxy{
+			&context,
+			proxy,
+			KubeConfig,
+		}
+	}
+}
+
+// watchForKubeConfigChanges watches for changes in the kubeconfig file and
+// refreshes the config when it changes.
+func watchForKubeConfigChanges(config *HeadlampConfig) {
+	path := config.kubeConfigPath
+
+	const tickerDuration = 10 * time.Second
+
+	log.Println("setting up watcher for kubeconfig changes", path)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Println("Error watching for kube config changes:", err)
+		return
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
+
+	ticker := time.NewTicker(tickerDuration)
+
+	err = watcher.Add(path)
+	if err != nil {
+		log.Printf("Couldn't add %s to watcher: %v", path, err)
+	}
+
+	go func() {
+		for {
+			select {
+			// when the kubeconfig file is removed the watcher stops listening
+			// so this is a workaround to check if the file exists and add it to the watcher
```

Comment:
This cant be added when using the `fsnotify.Rename` or `fsnotify.Remove` event because the file is automatically removed from the watcher when the file is removed so we won't get those events in the first place.

Reply:
So when does these triggers happen? Maybe they're happening for directories only?

Reply:
Discussed this privately and updated the comment to make it clear.